load_confounds/parser.py

```diff
@@ -76,34 +77,37 @@ def _load_high_pass(confounds_raw):
     high_pass_params = _find_confounds(confounds_raw, ["cosine"])
     return confounds_raw[high_pass_params]
 
-
-def _label_compcor(confounds_raw, compcor_suffix, n_compcor):
+def _label_compcor(confounds_json, compcor_suffix, n_compcor, acompcor_mask):
```

Comment:
this function has many if statements, making the flow of execution very hard to follow. I made some suggestions to try and simplify.